hkmc2/shared/src/main/scala/hkmc2/utils/Scope.scala

```diff
@@ -96,7 +96,7 @@ class Scope
             case t: syntax.Tree.TypeDef => t.head.toLoc
           .flatten.orElse(l.toLoc)
         case other => other.toLoc
-      raise(InternalError(msg"Not in scope: ${l.toString} (${l.getClass.toString})" -> loc :: Nil,
+      raise(ErrorReport(msg"No definition found in scope for: ${l.toString} (${l.getClass.toString})" -> loc :: Nil,
```

Comment:
Tip: error reports should be user-readable and shouldn't include implementation details like symbol identifiers and runtime class names. So I'd use:
```suggestion
      raise(ErrorReport(msg"No definition found in scope for '${l.nme}'" -> loc :: Nil,
        extraInfo = Some(l -> l.getClass),
        source = Diagnostic.Source.Compilation))
```
since `extraInfo` is the parameter for additional debugging info, which will gives us:
```fs
fun f: Int

f
//│ FAILURE: Unexpected compilation error
//│ FAILURE LOCATION: lookup_! (Scope.scala:99)
//│ FAILURE INFO: Tuple2:
//│   _1 = member:f
//│   _2 = class hkmc2.semantics.BlockMemberSymbol
//│ ╔══[COMPILATION ERROR] No definition found in scope for 'f'
//│ ║  l.12: 	fun f: Int
//│ ╙──      	    ^^^^^^
//│ FAILURE: Unexpected runtime error
//│ FAILURE LOCATION: mkQuery (JSBackendDiffMaker.scala:149)
//│ ═══[RUNTIME ERROR] ReferenceError: f is not defined
```